File: internal/realtimegateway/transports/status/health_router.go
package status

import (
	"net/http"

	gin "github.com/gin-gonic/gin"
)

func ConfigureHealthRouter(router gin.IRouter, isHealthy func() bool) {
	router.GET("/healthz", func(ctx *gin.Context) {
		if !isHealthy() {
			ctx.Status(http.StatusServiceUnavailable)
			return
		}

		ctx.Status(http.StatusOK)
	})
}

File: internal/yjsworker/transports/status/health_router.ts
import type { Hono } from "hono";

export function configureHealthRouter(
  app: Hono,
  isHealthy: () => boolean
): void {
  app.get("/healthz", context => {
    if (!isHealthy()) {
      return context.body(null, 503);
    }

    return context.body(null, 200);
  });
}
